Code/Framework/AzCore/AzCore/DOM/DomValue.cpp

```diff
@@ -0,0 +1,1305 @@
+            return aznumeric_cast<uint64_t>(AZStd::get<double>(m_value));
+        }
+        AZ_Assert(false, "AZ::Dom::Value: Called GetInt on a non-numeric type");
+        return {};
+    }
+
+    void Value::SetUint64(uint64_t value)
+    {
+        m_value = value;
+    }
+
+    uint32_t Value::GetUint32() const
+    {
+        return aznumeric_cast<uint32_t>(GetUint64());
+    }
+
+    void Value::SetUint32(uint32_t value)
+    {
+        m_value = aznumeric_cast<uint64_t>(value);
+    }
+
+    bool Value::GetBool() const
+    {
+        if (IsBool())
+        {
+            return AZStd::get<bool>(m_value);
+        }
+        AZ_Assert(false, "AZ::Dom::Value: Called GetBool on a non-bool type");
+        return {};
+    }
+
+    void Value::SetBool(bool value)
+    {
+        m_value = value;
+    }
+
+    double Value::GetDouble() const
+    {
+        switch (m_value.index())
+        {
+        case GetTypeIndex<int64_t>():
+            return aznumeric_cast<double>(AZStd::get<int64_t>(m_value));
+        case GetTypeIndex<uint64_t>():
+            return aznumeric_cast<double>(AZStd::get<uint64_t>(m_value));
+        case GetTypeIndex<double>():
+            return AZStd::get<double>(m_value);
+        }
+        AZ_Assert(false, "AZ::Dom::Value: Called GetInt on a non-numeric type");
+        return {};
+    }
+
+    void Value::SetDouble(double value)
+    {
+        m_value = value;
+    }
+
+    float Value::GetFloat() const
+    {
+        return aznumeric_cast<float>(GetDouble());
+    }
+
+    void Value::SetFloat(float value)
+    {
+        m_value = aznumeric_cast<double>(value);
+    }
+
+    void Value::SetString(SharedStringType sharedString)
+    {
+        m_value = sharedString;
+    }
+
+    AZStd::string_view Value::GetString() const
+    {
+        switch (m_value.index())
+        {
+        case GetTypeIndex<AZStd::string_view>():
+            return AZStd::get<AZStd::string_view>(m_value);
+        case GetTypeIndex<SharedStringType>():
+            {
+                auto& buffer = *AZStd::get<SharedStringType>(m_value);
+                return { buffer.data(), buffer.size() };
+            }
+        case GetTypeIndex<ShortStringType>():
+            {
+                const ShortStringType& shortString = AZStd::get<ShortStringType>(m_value);
+                return { shortString.data(), shortString.size() };
+            }
+        }
+        AZ_Assert(false, "AZ::Dom::Value: Called GetString on a non-string type");
+        return {};
+    }
+
+    size_t Value::GetStringLength() const
+    {
+        return GetString().size();
+    }
+
+    void Value::SetString(AZStd::string_view value)
+    {
+        if (value.size() <= ShortStringSize)
+        {
+            ShortStringType buffer;
+            buffer.resize_no_construct(value.size());
+            memcpy(buffer.data(), value.data(), value.size());
+            m_value = buffer;
+        }
+        m_value = value;
```

Comment:
Heh woops, the intent was to always store small strings, but since the reference is guaranteed longevity I'll go ahead and make the copy only happen for short strings that are designed to be stored since we're not trying to prevent an allocation.